src/main/java/org/gridsuite/directory/notification/server/DirectoryNotificationWebSocketHandler.java

```diff
@@ -154,18 +152,35 @@ private Flux<WebSocketMessage> heartbeatFlux(WebSocketSession webSocketSession)
                 .pingMessage(dbf -> dbf.wrap((webSocketSession.getId() + "-" + n).getBytes(StandardCharsets.UTF_8))));
     }
 
+    public Flux<WebSocketMessage> receive(WebSocketSession webSocketSession) {
+        return webSocketSession.receive()
+                .doOnNext(webSocketMessage -> {
+                    try {
+                        String wsPayload = webSocketMessage.getPayloadAsText();
+                        //if it's not the heartbeat
+                        if (!(wsPayload.startsWith(webSocketSession.getId() + "-"))) {
```

Comment:
are you sure there is not better way to detect ping messages ?

Reply:
didn't find another way, open to suggestion

Reply:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/reactive/socket/WebSocketMessage.html#getType()
 ?

Reply:
fixed